plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -214,6 +216,10 @@ public class ClickHouseClient
 
         Domain simplifiedDomain = domain.simplify(getDomainCompactionThreshold(session));
         if (!simplifiedDomain.getValues().isDiscreteSet()) {
+            // Push down inequality predicate if domain was not simplified
+            if (domain == simplifiedDomain) {
+                return FULL_PUSHDOWN.apply(session, domain);
+            }
```

Comment:
Can this be a prep commit ?

Reply:
This enables pushdown for inequalities. 
And it's very tricky to chop this code to separate commits within this PR: either one or another preexisting base test fails because it lacks some functionality. So either it would require eventually not needed manipulations with tests (finally all they will have to be removed), or to have it as it is.

Reply:
https://github.com/trinodb/trino/pull/23516